core/arch/arm/include/sm/optee_smc.h

```diff
@@ -422,6 +422,32 @@
 #define OPTEE_SMC_VM_DESTROYED \
 	OPTEE_SMC_FAST_CALL_VAL(OPTEE_SMC_FUNCID_VM_DESTROYED)
 
+/*
+ * Query OP-TEE about number of supported threads
+ *
+ * Normal World OS or Hypervisor issues this call to find out how much threads
+ * OP-TEE support. In other hand, how many standard calls can be issued in
+ * parallel before OP-TEE will return OPTEE_SMC_RETURN_ETHREAD_LIMIT.
+ *
+ * Call requests usage:
+ * a0	SMC Function ID, OPTEE_SMC_GET_THREAD_COUNT
+ * a1-6 Not used
+ * a7	Hypervisor Client ID register
```

Comment:
Is this value used? What value should be passed if caller is not an hypervisor?

Reply:
No, it is not used right now. While it is theoretically possible to have a different configuration for a different virtual guests, I can't see how it can be implemented right now.

But it is described in SMC Calling Convention. I can remove this, if you think that it is not needed.

Reply:
If it's "generic stuff" from SMCC I guess you can leave it, it's OK.